File: public/hooks/convert_arg.py
# .\public\hooks\convert_arg.py
import ast

def convert_arg(value):
    """
    尝试将字符串转换为合适的 Python 类型：
    1. 如果是 "True" 或 "False"（不区分大小写），转换为 bool
    2. 尝试用 ast.literal_eval 安全解析 Python 字面量（列表、字典等）
    3. 如果能转换为整数，则返回 int
    4. 如果能转换为浮点数，则返回 float
    5. 否则返回原字符串
    """
    lower_val = value.lower()
    if lower_val == "true":
        return True
    if lower_val == "false":
        return False

    # 尝试用 ast.literal_eval 解析 Python 字面量
    try:
        return ast.literal_eval(value)
    except (ValueError, SyntaxError):
        pass

    # 尝试整数
    try:
        return int(value)
    except ValueError:
        pass

    # 尝试浮点数
    try:
        return float(value)
    except ValueError:
        pass

    # 默认返回字符串
    return value

File: public/hooks/convertArg.js
// .\public\hooks\convertArg.js
/**
 * 尝试将字符串转换为合适的 JavaScript 类型（保留 "true"/"false" 为字符串）
 * @param {string} value - 原始字符串参数
 * @returns {any} 转换后的值
 */
function convertArg(value) {
  // 不再自动将 "true"/"false" 转换为布尔值，保持为字符串

  // 尝试转换为数字（整数或浮点数）
  if (/^-?\d+$/.test(value)) {
    return parseInt(value, 10);
  }
  if (/^-?\d*\.\d+$/.test(value)) {
    return parseFloat(value);
  }

  // 尝试解析 JSON（支持双引号格式）
  try {
    return JSON.parse(value);
  } catch {
    // 忽略错误，继续尝试其他方法
  }

  // 尝试将单引号替换为双引号后解析 JSON（适配 Python 风格字面量）
  try {
    const jsonCompatible = value.replace(/'/g, '"');
    return JSON.parse(jsonCompatible);
  } catch {
    // 忽略错误
  }

  // 默认返回原字符串
  return value;
}

export default convertArg;
